File: 2021/day1.py
def main():
    with open("day1.input") as f:
        numbers = list(map(int, f))

    print(part1(numbers))
    print(part2(numbers))


def part1(numbers):
    total = 0
    prev = numbers[0]
    for next_n in numbers[1:]:
        if next_n > prev:
            total += 1

        prev = next_n

    return total


def part2(numbers):
    total = 0
    prev = numbers[0] + numbers[1] + numbers[2]

    for i in range(len(numbers[1:])):
        if i + 3 > len(numbers):
            break

        next_n = numbers[i] + numbers[i + 1] + numbers[i + 2]
        if next_n > prev:
            total += 1

        prev = next_n
    return total


if __name__ == "__main__":
    main()

File: 2021/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	var (
		file    *os.File
		scanner *bufio.Scanner
		numbers []int
	)

	file, _ = os.Open("day1.input")
	defer file.Close()

	scanner = bufio.NewScanner(file)
	for scanner.Scan() {
		i, _ := strconv.Atoi(scanner.Text())
		numbers = append(numbers, i)
	}

	fmt.Println(partOne(numbers))
	fmt.Println(partTwo(numbers))
}

func partOne(numbers []int) int {
	var (
		total int
		prev  int
	)
	prev = numbers[0]
	for _, next := range numbers[1:] {
		if next > prev {
			total++
		}
		prev = next
	}
	return total
}

func partTwo(numbers []int) int {
	var (
		prev  int = numbers[0] + numbers[1] + numbers[2]
		total int
	)
	for i, next := range numbers[1:] {
		if i+3 > len(numbers) {
			break
		}

		next = numbers[i] + numbers[i+1] + numbers[i+2]
		if next > prev {
			total++
		}
		prev = next
	}
	return total
}
